docs/guides/hybrid-graphics.md

```diff
@@ -77,6 +79,65 @@ If you experience system freezes, then the laptop's fans becoming loud, before t
 
 `glxinfo | grep "OpenGL renderer"` should show an Intel GPU. Running programs with `DRI_PRIME=1` will make them render on your AMD GPU (some things do this automatically). You will get more battery time now as your AMD GPU can be turned off when not needed.
 
+### NixOS
+
+OBS! These instructions assumes a systemd bootloader. If using something else you will need to make some minor tweaks to assure the .efi file ends up in correct dir and gets set as default.
+
+1.  Create a new file in `/etc/nixos/` called `hybrid_graphics.nix`:
+
+    ```plain
+    { config, pkgs, ... }:
+    let
+    hybrid-graphics = pkgs.stdenv.mkDerivation rec {
+        name = "hybrid-graphics-1.0";
+
+        src = pkgs.fetchFromGitHub {
+        owner = "Redecorating";
+        repo = "apple_set_os-loader";
+        rev = "r33.9856dc4";
+        sha256 = "hvwqfoF989PfDRrwU0BMi69nFjPeOmSaD6vR6jIRK2Y=";
+        };
+
+        buildInputs = [ pkgs.gnu-efi ];
+
+        buildPhase = ''
+        substituteInPlace Makefile --replace "/usr" '$(GNU_EFI)'
+        export GNU_EFI=${pkgs.gnu-efi}
+        make
+        '';
+
+        installPhase = ''
+        install -D bootx64_silent.efi $out/bootx64.efi
+        '';
+    };
+    in
+    {
+    system.activationScripts.hybrid-graphics = {
+        text = ''
+        mkdir -p /boot/efi/EFI/hybrid_graphics
+        cp ${hybrid-graphics}/bootx64.efi /boot/efi/EFI/hybrid_graphics/bootx64.efi
+        cp /boot/efi/EFI/BOOT/BOOTX64.EFI /boot/efi/EFI/BOOT/bootx64_original.efi
+        '';
+    };
+
+    environment.etc."modprobe.d/apple-gmux.conf".text = ''
+    # Enable the iGPU by default if present
+    options apple-gmux force_igd=y
+    '';
+
+    environment.systemPackages = with pkgs; [ hybrid-graphics ];
+    }
+    ```
+
+2.  Add it to the imports in `/etc/nixos/configuration.nix`.
+
+3.  ```sh
+    sudo nixos-rebuild switch
+    ```
+4.  Use `efibootmgr` to create a new default boot entry point: `/boot/efi/EFI/hybrid_graphics/bootx64.efi` (see https://nixos.wiki/wiki/Bootloader for more details). It will then automatically load `/boot/efi/EFI/BOOT/bootx64_original.efi` which is a copy of the main loader (in this case systemd), so carefully check to make sure that file has been generated correctly. If not then adjust the relevant lines in `hybrid_graphics.nix`.
```

Comment:
It needs to be in the default boot dir (`/boot/efi/EFI/BOOT/bootx64.efi`) for apple firmware to use it.